package-system/AWSGameLiftServerSDK/build_package_image.py

```diff
@@ -0,0 +1,289 @@
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--platform-name',
+        required=True,
+        choices=PACKAGE_PLATFORM_OPTIONS
+    )
+    
+    args = parser.parse_args()
+    global PACKAGE_PLATFORM
+    PACKAGE_PLATFORM = args.platform_name
+
+def prepare_working_directory() -> WorkingDirectoryInfo:
+    # root directory
+    root_directory: pathlib.Path = PACKAGE_BASE_PATH.joinpath("temp")
+    if root_directory.is_dir():
+        delete_folder(root_directory)
+    
+    # source and build directory
+    source_directory: pathlib.Path = \
+        root_directory.joinpath(f"GameLift-SDK-Release-{GAMELIFT_SERVER_SDK_RELEASE_VERSION}/GameLift-Cpp-ServerSDK-{PACKAGE_VERSION.split('-')[0]}")
+    build_directory: pathlib.Path = root_directory.joinpath("build")
+    build_directory.mkdir(parents=True)
+    
+    # build output and libs output directory
+    build_output_directory: pathlib.Path = PACKAGE_ROOT_PATH.joinpath(f"{PACKAGE_NAME}-{PACKAGE_PLATFORM}")
+    if build_output_directory.is_dir():
+        delete_folder(build_output_directory)
+    
+    build_libs_output_directory: pathlib.Path = build_output_directory.joinpath(f"{PACKAGE_NAME}")
+    build_libs_output_directory.mkdir(parents=True)
+    
+    return WorkingDirectoryInfo(root_directory, source_directory, build_directory, build_output_directory, build_libs_output_directory)
+
+def download_gamelift_server_sdk(working_directory: WorkingDirectoryInfo) -> None:
+    # download sdk from url
+    gamelift_sdk_zip_file: str = working_directory.root_path.joinpath("gamelift_server_sdk.zip").resolve()
+    with urllib.request.urlopen(GAMELIFT_SERVER_SDK_DOWNLOAD_URL) as response, open(gamelift_sdk_zip_file, 'wb') as out_file:
+        shutil.copyfileobj(response, out_file)
+    
+    # unzip sdk contents
+    with zipfile.ZipFile(gamelift_sdk_zip_file, "r") as f:
+        f.extractall(working_directory.root_path.resolve())
+
+def apply_patch_on_sdk_source(working_directory: WorkingDirectoryInfo) -> None:
+    patch_file: str = PACKAGE_BASE_PATH.joinpath(f"{PACKAGE_NAME}-{PACKAGE_VERSION.split('-')[0]}-{PACKAGE_PLATFORM}.patch").resolve()
+    source_folder: str = working_directory.source_path.resolve()
+    git_result = subprocess.run(subp_args(["git","init"]), 
+                                shell=True,
+                                capture_output=True,
+                                cwd=source_folder)
+    if git_result.returncode != 0:
+        raise Exception(f"Error git init sdk source: {str(git_result.stderr)}")
+    git_result = subprocess.run(subp_args(["git","add","."]), 
+                                shell=True,
+                                capture_output=True,
+                                cwd=source_folder)
+    if git_result.returncode != 0:
+        raise Exception(f"Error git add sdk source: {str(git_result.stderr)}")
+    git_result = subprocess.run(subp_args(["git","commit","-m","temp"]),
```

Comment:
Curious: What are the changes we're committing here?

Reply:
it is for apply patch file if the directory is not a git workspace initially